app/src/main/java/com/github/h3lp3rs/h3lp/signin/SignInActivity.kt

```diff
@@ -53,126 +109,149 @@ class SignInActivity : AppCompatActivity() {
-        // Initialize Firebase Auth
-        findViewById<ImageButton>(R.id.signInButton).setOnClickListener{
-            launchSignIn()
-        }
-
-        // Continue without sign in
-        findViewById<TextView>(R.id.noSignInText).setOnClickListener {
-            username = GUEST_USER
-            disableOnlineSync()
-            checkToSAndLaunchIfNotAcceptedElseMain()
-        }
-
-        // Sign in local storage doesn't need online sync
-        USER_SIGNED_IN = getString(R.string.KEY_USER_SIGNED_IN)
-        USER_UID = getString(R.string.KEY_USER_UID)
-        USER_NAME = getString(R.string.KEY_USER_NAME)
-        // Check if the user is already signed in
-        offlineCheckIfSignedIn()
-    }
 
     /**
      * Initialize client and launch the sign in request
      */
-    private fun launchSignIn(){
-        signInClient = SignIn.get()
-        val signInIntent = signInClient.signIn(this)
-        resultLauncher.launch(signInIntent)
+    private fun launchSignIn() {
+        if (checkUsernameField()) {
+            signInClient = SignIn.get()
+            val signInIntent = signInClient.signIn(this)
+            resultLauncher.launch(signInIntent)
+        } else {
+            displayMessage(
+                findViewById<View>(android.R.id.content).rootView,
+                getString(R.string.username_error_field_msg)
+            )
+        }
     }
 
     // Handle sign in request result
-    private val resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult())
-        { result -> authenticateUser(result,this) }
+    private val resultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            authenticateUser(
+                result, this
+            )
+        }
 
     /**
      * Authenticate user
      *
      * @param result sign in intent result containing the user account
      * @param activity current activity
      */
-    fun authenticateUser(result: ActivityResult, activity: Activity){
-        signInClient.authenticate(result, activity)
-            ?.addOnCompleteListener(activity) { task ->
-                if (task.isSuccessful) {
-                    userUid = signInClient.getUid()
-                    // Only get the first name for privacy reasons
-                    username = getInstance().currentUser?.displayName?.substringBefore(" ")
-
-                    saveAuthentication()
-
-                    // Enable online sync for meaningful storages:
-                    SKILLS.setOnlineSync(true)
-                    MEDICAL_INFO.setOnlineSync(true)
-                    USER_COOKIE.setOnlineSync(true)
-
-                    checkToSAndLaunchIfNotAcceptedElseMain()
-                }
+    fun authenticateUser(result : ActivityResult ?, activity : Activity) {
+        signInClient.authenticate(result, activity)?.addOnCompleteListener(activity) { task ->
+            if (task.isSuccessful) {
```

Comment:
This will be executed if we sign in anonymously or using our google account. In the first case, it's authenticate from AnonymousSignInAdapter which will be called, and authenticate from GoogleSignInAdapter for the second case. If we choose the third option which is to continue without signing up this will not be executed :)